frontend/src/app/testResults/TestResultsList.tsx

```diff
@@ -557,135 +529,180 @@ export const resultsCountQuery = gql`
+    history.push({
+      pathname: "/results/1",
+      search: qs.stringify({
+        facility: activeFacilityId,
+        ...filterParams,
+        ...params,
+      }),
+    });
+  };
+
+  const setFilterParams = (key: keyof FilterParams) => (val: string | null) => {
+    filter({ [key]: val });
+  };
+
+  const refetch = () => history.go(0);
+
+  const clearFilterParams = () =>
+    history.push({
+      pathname: "/results/1",
+      search: qs.stringify({ facility: activeFacilityId }),
+    });
 
   const entriesPerPage = 20;
-  const pageNumber = props.page || 1;
+  const pageNumber = props.pageNumber || 1;
 
-  const queryVariables: {
-    patientId?: string;
+  const resultsQueryVariables: {
+    patientId?: string | null;
     facilityId: string;
-    result?: string;
-    role?: string;
-    startDate?: string;
-    endDate?: string;
+    result?: string | null;
+    role?: string | null;
+    startDate?: string | null;
+    endDate?: string | null;
     pageNumber: number;
     pageSize: number;
   } = {
     facilityId: activeFacilityId,
     pageNumber: pageNumber - 1,
     pageSize: entriesPerPage,
+    ...filterParams,
   };
-
   const countQueryVariables: {
-    patientId?: string;
+    patientId?: string | null;
     facilityId: string;
-    result?: string;
-    role?: string;
-    startDate?: string;
-    endDate?: string;
-  } = { facilityId: activeFacilityId };
-
-  if (selectedPatientId) {
-    queryVariables.patientId = selectedPatientId;
-    countQueryVariables.patientId = selectedPatientId;
-  }
-
-  if (resultFilter) {
-    queryVariables.result = resultFilter;
-    countQueryVariables.result = resultFilter;
-  }
-
-  if (roleFilter) {
-    queryVariables.role = roleFilter;
-    countQueryVariables.role = roleFilter;
-  }
-
-  if (startDateFilter) {
-    queryVariables.startDate = startDateFilter;
-    countQueryVariables.startDate = startDateFilter;
-  }
-
-  if (endDateFilter) {
-    queryVariables.endDate = endDateFilter;
-    countQueryVariables.endDate = endDateFilter;
-  }
+    result?: string | null;
+    role?: string | null;
+    startDate?: string | null;
+    endDate?: string | null;
+  } = {
+    facilityId: activeFacilityId,
+    ...filterParams,
+  };
 
-  const {
-    data: totalResults,
-    loading: loadingTotalResults,
-    error,
-    refetch: refetchCount,
-  } = useQuery(resultsCountQuery, {
+  const count = useQuery(resultsCountQuery, {
+    fetchPolicy: "no-cache",
     variables: countQueryVariables,
+  });
+  const results = useQuery(testResultQuery, {
     fetchPolicy: "no-cache",
+    variables: resultsQueryVariables,
   });
 
   if (!activeFacilityId) {
     return <div>"No facility selected"</div>;
   }
 
-  if (error) {
-    throw error;
+  if (results.error || count.error) {
+    throw results.error || count.error;
   }
 
-  const totalEntries = totalResults?.testResultsCount || 0;
+  const totalEntries = count.data?.testResultsCount || 0;
 
   return (
-    <QueryWrapper<Props>
```

Comment:
I think the only thing we lose here from _not_ using QueryWrapper is the AppInsights front end tracking:

```
  const appInsights = useAppInsightsContext();
  const trackAction = useTrackEvent(appInsights, "User Action", {});
```

**suggestion:** Will you add this feature back in?

Reply:
oh i didn't know that was in there, makes sense, I'll add it back